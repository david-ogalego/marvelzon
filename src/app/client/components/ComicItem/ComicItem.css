.item {
    margin: 10px 54px 54px 54px;
}

.containerImage {
    height: 149px;
    width: 216px;
    position: relative;
    padding-top: 133px;
    margin-bottom: 120px;
}

.details {
    position: absolute;
    top: 0;
    z-index: -1;
}

.image {
    margin-top: -120px;
}

.image:hover {
    transform: translateY(120px);
    transition: 1s ease-in-out;
}