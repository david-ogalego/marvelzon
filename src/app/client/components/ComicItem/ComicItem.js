import React from 'react';
import { Link } from 'react-router-dom';

const ComicItem = function(props) {
    return (
        <Link key={props.videoId} to={`/detail/${props.id}`} >
            <h3>{props.title}</h3>
            Date: <label>{props.date}</label>
            Price: <label>{props.price}</label>
            <img alt={props.title} src={props.image} />
        </Link>
    );
};

export default ComicItem;