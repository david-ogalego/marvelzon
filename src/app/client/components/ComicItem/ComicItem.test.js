import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ComicItem from './ComicItem';
import MarvelLogo from '../../images/MarvelLogo.png';

test('Snapshot ComicDetail rendered with comic', () => {
    const treeComicDetail = renderer.create(
        <MemoryRouter>
            <ComicItem
                id={0}
                title={'X-Men'}
                date={'2029-12-31T00:00:00-0500'}
                price={200}
                image={MarvelLogo}
                pageCount={10}
                description={'Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen.'}
                url={'http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b'} />
        </MemoryRouter>        
        ).toJSON();
    expect(treeComicDetail).toMatchSnapshot();
});

test('Snapshot ComicDetail rendered with comic and without price', () => {
    const treeComicDetail = renderer.create(
        <MemoryRouter>
            <ComicItem
                id={0}
                title={'X-Men'}
                date={'2029-12-31T00:00:00-0500'}
                price={0}
                image={MarvelLogo}
                pageCount={10}
                description={'Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen.'}
                url={'http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b'} />
        </MemoryRouter>        
        ).toJSON();
    expect(treeComicDetail).toMatchSnapshot();
});