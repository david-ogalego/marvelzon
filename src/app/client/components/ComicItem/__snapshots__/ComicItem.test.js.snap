// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ComicDetail rendered with comic 1`] = `
<a
  className="item"
  href="/detail/0"
  onClick={[Function]}
>
  <div
    className="containerImage"
  >
    <img
      alt="X-Men"
      className="image"
      src="test-file-stub"
    />
    <div
      className="containerDetails"
    >
      <div
        className="detail titleDetail"
      >
        X-Men
      </div>
      <div
        className="detail"
      >
        2029-12-31
      </div>
      <div
        className="detail"
      >
        Price: 
        200
      </div>
    </div>
  </div>
</a>
`;

exports[`Snapshot ComicDetail rendered with comic and without price 1`] = `
<a
  className="item"
  href="/detail/0"
  onClick={[Function]}
>
  <div
    className="containerImage"
  >
    <img
      alt="X-Men"
      className="image"
      src="test-file-stub"
    />
    <div
      className="containerDetails"
    >
      <div
        className="detail titleDetail"
      >
        X-Men
      </div>
      <div
        className="detail"
      >
        2029-12-31
      </div>
    </div>
  </div>
</a>
`;
